### PR TITLE
Fix #134

### LINKED ISSUE #134: Error in edgeR analysis: library size of zero detected when executing DGEList function
I have checked the following places for your error:

- [ ] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [ ] [nf-core/smrnaseq pipeline documentation](https://nf-co.re/nf-core/smrnaseq/usage)

## Description of the bug
I have tried executing nf-core/smrnna workflow, up until the edgeR step the following error shows up
<Command exit status:
  1

Command output:
  $mature
   [1] "SAGCFN_22_00196.mature.stats" "SAGCFN_22_00194.mature.stats"
   [3] "SAGCFN_22_00188.mature.stats" "SAGCFN_22_00190.mature.stats"
   [5] "SAGCFN_22_00189.mature.stats" "SAGCFN_22_00193.mature.stats"
   [7] "SAGCFN_22_00186.mature.stats" "SAGCFN_22_00187.mature.stats"
   [9] "SAGCFN_22_00197.mature.stats" "SAGCFN_22_00191.mature.stats"
  [11] "SAGCFN_22_00195.mature.stats" "SAGCFN_22_00192.mature.stats"
  
  $hairpin
   [1] "SAGCFN_22_00187.hairpin.stats" "SAGCFN_22_00192.hairpin.stats"
   [3] "SAGCFN_22_00191.hairpin.stats" "SAGCFN_22_00188.hairpin.stats"
   [5] "SAGCFN_22_00186.hairpin.stats" "SAGCFN_22_00195.hairpin.stats"
   [7] "SAGCFN_22_00189.hairpin.stats" "SAGCFN_22_00197.hairpin.stats"
   [9] "SAGCFN_22_00196.hairpin.stats" "SAGCFN_22_00190.hairpin.stats"
  [11] "SAGCFN_22_00193.hairpin.stats" "SAGCFN_22_00194.hairpin.stats"

Command error:
  Loading required package: limma
  Loading required package: edgeR
  Loading required package: statmod
  Loading required package: data.table
  Loading required package: gplots
  
  Attaching package: ‘gplots’
  
  The following object is masked from ‘package:stats’:
  
      lowess
  
  Error in cpm.default(y$counts, lib.size = lib.size, log = log, prior.count = prior.count) : 
    library sizes should be finite and non-negative
  Calls: cpm -> cpm.DGEList -> cpm.default
  In addition: Warning message:
  In DGEList(counts = data, genes = rownames(data)) :
    library size of zero detected
  Execution halted

Work dir:
  /cancer/storage/SAGC/projects/SAGCQA0237-RainerHaberberger/raw_data/work/9a/6eedaca4a80898bec9761a56b5a926

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line>

## Steps to reproduce

Steps to reproduce the behaviour:

1. Command line:
 'nextflow run nf-core/smrnaseq --input '*.fastq.gz' --genome GRCh38 -profile sahmri --custom_config_base 'https://raw.githubusercontent.com/sagc-bioinformatics/configs/sahmri' --mature ../miRBase/mature.fa.gz --hairpin ../miRBase/hairpin.fa.gz -resume -r 1.1.0'

2.. See error: 
<  Error in cpm.default(y$counts, lib.size = lib.size, log = log, prior.count = prior.count) : 
    library sizes should be finite and non-negative
  Calls: cpm -> cpm.DGEList -> cpm.default
  In addition: Warning message:
  In DGEList(counts = data, genes = rownames(data)) :
    library size of zero detected
  Execution halted>

## Log files

Have you provided the following extra information/files:

- [ ] The command used to run the pipeline
- [ ] The `.nextflow.log` file 

## System

- Hardware: 
- Executor: 

## Nextflow Installation

- Version: 

## Container engine

- Engine: <Singularity>
- Image tag: <nf-core 1.1.0>

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

This only adds a simple bugfix for cases where a _sample_ is entirely zero for all genes that are left after filtering out for zero expression genes in the matrix. Fixes #134 - tests with a broken case are running 👍🏻

### CODE REVIEW

**Comment by apeltzer** [comment]
Confirmed and tested that this indeed fixes the issue 👍🏻

**Review by jfy133 — APPROVED** [review]

File: bin/edgeR_miRBase.r [hunk 1]
```diff
@@ -41,11 +41,14 @@ for (i in 1:2) {
     # Remove genes with 0 reads in all samples
     row_sub = apply(data, 1, function(row) all(row ==0 ))
     # Only subset if at least one sample is remaining
-    nr_keep <- table(row_sub)
-    nr_keep <- sum(nr_keep == TRUE)
+    nr_keep <- sum(row_sub)
     if (nr_keep > 0){
         data<-data[!row_sub,]
     }
+
```
Comment (jfy133) [inline bin/edgeR_miRBase.r:48]:
```suggestion

```

Comment (jfy133) [reply bin/edgeR_miRBase.r:48]: To fix linting

**Comment by jfy133** [comment]
> Confirmed and tested that this indeed fixes the issue 👍🏻

I'll trust you there ;).

**Pull request merged**